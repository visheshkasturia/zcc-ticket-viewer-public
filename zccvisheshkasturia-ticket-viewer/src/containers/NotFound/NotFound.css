.not-found{
  margin-top: 100px;
  margin-left: auto;
  margin-right: auto;
  width: 90vw;
  border: solid 1px black;
  border-radius: 10px;
}

.h1-nf{
  text-align: center;
}

.img-gif{
  display: block;
  margin: auto;
  padding-top: 40px;
  border-radius: 10px;
  overflow: hidden;
}