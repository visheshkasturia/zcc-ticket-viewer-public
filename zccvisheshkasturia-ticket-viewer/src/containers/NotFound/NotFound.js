import React from 'react';
import './NotFound.css';
import c3pogif from '../../static/c3poomg.gif';

function NotFound() {
  return (
    <div className="not-found">
      <img data-testid="img-nf-test" className="img-gif" src={c3pogif} alt="Oh my goodness GIF" />
      <h1 data-testid="h1-nf-test" className="h1-nf">Looks like something went wrong and you landed in the wrong place . Quick try again!.</h1>
    </div>
  );
}

export default NotFound;
