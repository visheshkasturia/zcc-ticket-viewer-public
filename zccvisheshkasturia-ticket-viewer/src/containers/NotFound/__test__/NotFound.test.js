import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import NotFound from '../NotFound';

afterEach(cleanup);
describe('NotFound renders correctly', () => {
  it('<h1> has correct text', () => {
    const { getByTestId } = render(<NotFound />);
    expect(getByTestId('h1-nf-test')).toHaveTextContent('Looks like something went wrong and you landed in the wrong place . Quick try again!.');
  });

  it('Image has correct alt attibute', () => {
    const { getByTestId } = render(<NotFound />);
    expect(getByTestId('img-nf-test')).toHaveAttribute('alt', 'Oh my goodness GIF');
  });

  it('Matches snapshot', () => {
    const tree = renderer.create(<NotFound />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
