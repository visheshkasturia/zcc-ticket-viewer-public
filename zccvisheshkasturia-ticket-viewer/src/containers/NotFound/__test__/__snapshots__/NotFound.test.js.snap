// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound renders correctly Matches snapshot 1`] = `
<div
  className="not-found"
>
  <img
    alt="Oh my goodness GIF"
    className="img-gif"
    data-testid="img-nf-test"
    src="c3poomg.gif"
  />
  <h1
    className="h1-nf"
    data-testid="h1-nf-test"
  >
    Looks like something went wrong and you landed in the wrong place . Quick try again!.
  </h1>
</div>
`;
